core/src/androidTest/java/com/amplifyframework/core/DevMenuMainViewInstrumentationTest.java

```diff
@@ -0,0 +1,108 @@
+
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+
+import static androidx.test.espresso.Espresso.onView;
+
+/**
+ * Tests the navigation behavior for {@link DevMenuMainFragment}.
+ */
+public final class DevMenuMainViewInstrumentationTest {
+    // Amount of time (in milliseconds) to wait before starting
+    // a test, to allow any animations to complete.
+    private static final int START_TEST_DELAY_MS = 500;
+    // A navigation host controller for testing.
+    private TestNavHostController navHostController;
+
+    /**
+     * Go to the main screen of the developer menu.
+     */
+    @Before
+    public void resetView() {
+        navHostController = new TestNavHostController(ApplicationProvider.getApplicationContext());
+        navHostController.setGraph(R.navigation.dev_menu_nav_graph);
+        FragmentScenario<DevMenuMainFragment> mainMenuScenario =
+                FragmentScenario.launchInContainer(DevMenuMainFragment.class);
+        mainMenuScenario.onFragment(fragment -> Navigation.setViewNavController(fragment.requireView(),
+                navHostController));
+    }
+
+
+    /**
+     * Tests that the app is navigated to the environment information
+     * screen when the view environment information button is pressed.
+     */
+    @Test
+    public void testNavigationToEnvInfoScreen() {
+        testNavigationOnButtonPress(R.id.env_button, R.id.environment_fragment);
+    }
+
+    /**
+     * Tests that the app is navigated to the device information
+     * screen when the view device information button is pressed.
+     */
+    @Test
+    public void testNavigationToDeviceInfoScreen() {
+        testNavigationOnButtonPress(R.id.device_button, R.id.device_fragment);
+    }
+
+    /**
+     * Tests that the app is navigated to the logs screen when
+     * the view logs button is pressed.
+     */
+    @Test
+    public void testNavigationToLogsScreen() {
+        testNavigationOnButtonPress(R.id.logs_button, R.id.logs_fragment);
+    }
+
+    /**
+     * Tests that the app is navigated to the file issue screen
+     * when the file issue button is pressed.
+     */
+    @Test
+    public void testNavigationToFileIssueScreen() {
+        testNavigationOnButtonPress(R.id.file_issue_button, R.id.file_issue_fragment);
+    }
+
+    /**
+     * Test that the app is navigated to the destination with the given ID when
+     * the button with the given ID is pressed.
+     * @param buttonId ID of the button pressed
+     * @param destinationId ID of the navigation destination
+     */
+    private void testNavigationOnButtonPress(@RawRes int buttonId, @RawRes int destinationId) {
+        Sleep.milliseconds(START_TEST_DELAY_MS);
+        onView(ViewMatchers.withId(buttonId)).perform(ViewActions.click());
+        NavDestination curDestination = navHostController.getCurrentDestination();
+        Assert.assertNotNull(curDestination);
+        Assert.assertEquals(curDestination.getId(), destinationId);
```

Comment:
The predominant style in the repo is to use static imports for jUnit and Mockito utility code. That is, here, just `assertEquals`, `assertFoo`, `assertBar`, without the `Assert` class ref.